### DISCUSSION TITLE
Backend compilers users tend to use or prefer

### DISCUSSION THREAD

**Discussion by stefanos82** [edubart/nelua-lang#130] [General]
Dear all Nelua users,

Which backend compilers do you use with Nelua and why?

I personally use GCC almost always, for quick development though I use TCC; only once I had to use Clang and didn't like it as it was extremely slow on my computer for some reason.

What compilers do you use and what is your reasoning?

Cheers.

**Comment by ghost** [comment]
I only use GCC and haven't tried TCC, but i'm looking forward to try TCC.

I have not used Clang and will not use Clang because there's no reason to use Clang instead of GCC

**Reply by stefanos82** [reply]
You can compile it yourself to use the latest commit which is quite stable: https://repo.or.cz/w/tinycc.git

**Reply by ghost** [reply]
Okay :+1:

**Comment by edubart** [comment]
I mostly use the following:

* GCC on Linux, because generates the most efficient runtime code for me, compiles faster than Clang and also provides all features I need (code coverage, sanitizers, static analyzer, nice error messages)
* Clang on Windows (via MSVC or MSYS2), because it provides memory and undefined sanitizers, which are not available in MinGW GCC.
* Clang on MacOS, because it's the standard there, have all the tools I need and comes with XCode.
* GCC on all other exotic platforms, like Haiku, Solaris, FreeBSD, etc.

Worth mentioning:
* TCC, I use it sometimes when live coding something new that I want to iterate fast, because of its compile speed, but I often disable it when debugging or sanitizing my code.
* C2M from [MIR](https://github.com/vnmakarov/mir), I use it sometimes for fun and because I have future plans to use it in Nelua.
* Zig CC, because you can cross compile for different platforms with ease, in case you want to build an application for Linux from Windows and vice versa.
* Emscripten, I use to compile for WebAssembly and run applications in the browser of course.
* Musl GCC wrapper, I use it when I want to build portable command line applications for any Linux distribution, for example it is used to build Nelua releases for Linux.
* MingW32 GCC on Windows, because it is the GCC counterpart on Windows, so it supports most GCC features.
* CompCert C, because it's a formally verified optimizing C compiler, and you can expect it to generate very correct machine code when you think you may hit a GCC/Clang bug, I think it was created with the intent to run C code for control systems of airplanes for example, where you can't hit a bug because of the compiler otherwise the airplane may crash.

The right C compiler to use highly depends on your goal at the moment and platform.

**Reply by stefanos82** [reply]
Interesting list, very nice :+1:

**Reply by ghost** [reply]
indeed
